cmd/controlplane/external_webhooks.go

```diff
@@ -91,8 +93,41 @@ func (o *externalWebhooksServerOptions) run(ctx context.Context) error {
 		return fmt.Errorf("error registering warehouse by repo url indexer: %w", err)
 	}
 
+	err = cluster.GetFieldIndexer().IndexField(
+		ctx,
+		&kargoapi.ProjectConfig{},
+		indexer.ProjectConfigsByWebhookReceiverPathsField,
+		indexer.ProjectConfigsByWebhookReceiverPaths,
+	)
+	if err != nil {
+		return fmt.Errorf("error registering project configs by webhook receiver path indexer: %w", err)
+	}
+
+	go func() {
+		err = cluster.Start(ctx)
+	}()
+	if !cluster.GetCache().WaitForCacheSync(ctx) {
+		return fmt.Errorf("error waiting for cache to sync: %w", err)
+	}
+	if err != nil {
+		return fmt.Errorf("error starting cluster: %w", err)
+	}
+
 	srv := external.NewServer(serverCfg, cluster.GetClient())
-	l, err := net.Listen("tcp", fmt.Sprintf("%s:%s", o.Host, o.Port))
+
+	// The host MAY contain a port.
+	host, port, _ := net.SplitHostPort(o.Host)
+	if port == "" {
+		o.Logger.Info("No port specified in host, using fallback port", "port", o.Port)
+		port = o.Port
+	}
```

Comment:
After [this](https://github.com/akuity/kargo/pull/4000/files#r2105907239) is addressed, you can drop this. `LISTEN_ADDRESS` won't ever contain a port.